ko/기초 수학/유클리드 알고리즘.md

```diff
@@ -0,0 +1,92 @@
+# 유클리드 알고리즘
+
+## 문제
+
+양의 정수 $a$와 $b$의 최대공약수 $g = gcd(a, b)$를 구하시오. 여기서 $g$는 나머지 없이 $a$와 $b$ 모두 나누는 가장 큰 수로 정의한다.
+
+## 발상
+
+유클리드 알고리즘은 큰 수에서 작은 수를 빼면 두 수의 최대공약수가 변하지 않는다는 단순한 관찰을 기반한다:
+
+$a > b$인 상황에서 $g$는 $a$와 $b$의 최대공약수라고 하자.
+그렇다면 $g$는 $a$와 $b$를 나눌 수 있다. 따라서 $g$는 $a - b$도 나눌 수 있다. (분배법칙)
+
+$g'$를 $b$와 $a - b$의 최대공약수라 하자.
+
+$g' = g$ 귀류법:
+
+$g' < g$ 또는 $g' > g$이라고 가정하자.
+
+$g' < g$일 때, $g'$는 _최대_ 공약수가 될 수 없다.
+$g$도 $a - b$와 $b$의 공약수이기 때문이다.
+
+$g' > g$일 때, $g'$는 $b$와 $a - b$의 약수이다 -
+즉, $g'n = b$와 $g'm = a - b$로 표현할 수 있는 정수 $n, m$이 존재한다.
+따라서 $g'm = a - g'n \iff g'm + g'n = a \iff g'(m + n) = a$이다.
+이는 $g'$도 $a$의 약수이며, $g$가 $a$와 $b$의 최대공약수라는 초기 가정과 $g' > g$는 모순이다.
+
+## 구현
+
+실제 실행에서 속도를 높이기 위해 반복해서 뺄셈하는 대신 나머지 연산이 사용된다:
+$b$는 $a >= b$만큼 $a$에서 여러 번 반복해서 뺄 수 있다.
+이러한 뺄셈 후에는 $b$로 나눌 때 $a$의 나머지만 남게 된다.
+
+간단한 Lua 구현은 다음과 같다:
+
+```lua
```

Comment:
Pseudocode would be better here, IMO.
What do you folks think about this?

Reply:
The original file uses Lua; changing that is not in the scope of this translation PR. I've opened [a discussion](https://github.com/TheAlgorithms/Algorithms-Explanation/discussions/215) for this.